presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/MySqlShardDao.java

```diff
@@ -39,4 +40,14 @@
     @SqlBatch("INSERT IGNORE INTO deleted_shards (shard_uuid, delete_time)\n" +
             "VALUES (:shardUuid, CURRENT_TIMESTAMP)")
     void insertDeletedShards(@Bind("shardUuid") Iterable<UUID> shardUuids);
+
+    @SqlUpdate("DELETE FROM transactions\n" +
+            "WHERE end_time < :maxEndTime AND (\n" +
```

Comment:
I think we could simplify this to
```sql
AND successful IS NOT NULL
AND transaction_id NOT IN (...)
```
There shouldn't be any `created_shards` for successful transactions, but it probably doesn't hurt to check (not checking is an optimization rather than correctness).